Add nested checkbox guidance (#1245)
carbon-design-system/carbon-website: src/pages/components/checkbox/usage.mdx

</Column>
</Row>

### Nesting

Checkboxes can be nested when a parent and child relationship is needed. Users can either select an entire set of options or only a subset.

> Checking the parent checkbox will automatically check all its nested children checkboxes. Unchecking the parent checkbox will automatically uncheck all the children checkboxes.

Reviewer: ```suggestion
Checking the parent checkbox will automatically check all of the nested children checkboxes. Unchecking the parent checkbox will automatically uncheck all of the children checkboxes.
```

<Row>
<Column colLg={8}>

![Checking and unchecking the parent checkbox.](images/checkbox-usage-9a.png)

</Column>
</Row>

> Checking a child checkbox, if at least one other child checkbox remains unchecked, will automatically check the parent checkbox in the indeterminate state. Unchecking a child checkbox, when all other children checkboxes remain checked, will switch the parent checkbox from the default checked state to the indeterminate state.

Reviewer: ```suggestion
Checking a child checkbox, if at least one other child checkbox remains unchecked, will automatically put the parent checkbox into the indeterminate state. Unchecking a child checkbox, when all other children checkboxes remain checked, will switch the parent checkbox from the default checked state to the indeterminate state.
```
